### PR TITLE
Implement storage

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: network/src/storage.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+pub struct StorageImpl {
+    lock_file: Option<std::fs::File>,
+    path: String,
+}
+
+#[async_trait]
+impl Storage for StorageImpl {
+    async fn create(storage_directory: &str) -> Result<(), StorageError> {
+        let _ = fs::remove_dir_all(storage_directory).await;
```
Comment (lemonholic) [inline network/src/storage.rs:15]: This is not matched with the comment here: https://github.com/junha1/simperby/blob/3fcb5270c0249ee8764ff8eb890dd13aec51c095/network/src/primitives.rs#L10-L12

Either the implementation or the comment should be changed.

File: network/src/storage.rs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        Ok(())
+    }
+
+    async fn open(storage_directory: &str) -> Result<Self, StorageError>
+    where
+        Self: Sized,
+    {
+        let storage_directory_ = storage_directory.to_owned();
+        let file =
+            spawn_blocking(move || std::fs::File::open(format!("{}/lock", storage_directory_)))
```
Comment (lemonholic) [inline network/src/storage.rs:27]: Why do you use `std::fs` instead of tokio's?

File: network/src/storage.rs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        let mut file = fs::File::create(format!("{}/{}", self.path, name)).await?;
+        file.write_all(content.as_bytes()).await?;
+        Ok(())
+    }
+
+    async fn read_file(&self, name: &str) -> Result<String, StorageError> {
+        fs::read_to_string(format!("{}/{}", self.path, name)).await
+    }
+
+    async fn remove_file(&mut self, path: &str) -> Result<(), StorageError> {
```
Comment (lemonholic) [inline network/src/storage.rs:68]: use `name` instead of `path` here?

File: network/src/storage.rs (hunk 2 above)
Comment (junha1) [reply network/src/storage.rs:27]: `fs2` library is for file locking, but it does not support tokio. That's why I used `spawn_blocking()`.

Comment (lemonholic) [reply network/src/storage.rs:27]: ok

File: network/src/storage.rs (hunk 3 above)
Comment (junha1) [reply network/src/storage.rs]: fixed

**Review by lemonholic — APPROVED** [review]
good

**Pull request merged**